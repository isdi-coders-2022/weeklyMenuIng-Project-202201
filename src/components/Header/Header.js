import styled from "styled-components";
import BurgerComponent from "../BurgerComponent/BurgerComponent";
import Links from "../Links/Links";
import logo from "./logo.svg";

const HeaderContainer = styled.header`
  display: flex;
  height: 100px;
  align-items: center;
  background-color: #b3dee2;
  justify-content: space-around;

  & ul {
    display: flex;
    width: 100%;
    margin: 100px 0 10px 0;
    justify-content: space-between;
    align-items: center;
    padding: 0;
    list-style: none;
    margin: 0 10px;

    & li a {
      text-decoration: none;
      color: #000;
      margin: 0 8px;
    }
    @media (max-width: 530px) {
      display: none;
    }
  }

  @media (max-width: 650px) {
    font-size: 14px;
  }
  @media (max-width: 590px) {
    font-size: 13px;
  }
  @media (max-width: 560px) {
    font-size: 12px;
  }
`;

const Logo = styled.img`
  height: inherit;
`;

const Header = () => {
  return (
    <HeaderContainer>
      <Logo src={logo} alt="My weekly menu logo" />
      <Links />
      <BurgerComponent />
    </HeaderContainer>
  );
};
export default Header;
