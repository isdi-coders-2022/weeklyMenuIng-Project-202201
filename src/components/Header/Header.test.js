import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import userEvent from "@testing-library/user-event";
import Header from "./Header";
import TestRenderer from "react-test-renderer";

describe("Given a Header Component", () => {
  describe("When it's rendered", () => {
    test("Then the logo should display", () => {
      const altTextLogo = "My weekly menu logo";

      render(
        <BrowserRouter>
          <Header />
        </BrowserRouter>
      );

      const logo = screen.queryByAltText(altTextLogo);

      expect(logo).toBeInTheDocument();
    });
    test("Then it should match the snapshot", () => {
      const header = TestRenderer.create(
        <BrowserRouter>
          <Header />
        </BrowserRouter>
      ).toJSON();

      expect(header).toMatchSnapshot();
    });
  });

  describe("Wher it's rendered and cilcked on the burger menu", () => {
    test("Then the burger menu should have class active", () => {
      render(
        <BrowserRouter>
          <Header />
        </BrowserRouter>
      );

      const burger = screen.queryByTitle("Navigation Menu");
      userEvent.click(burger);

      expect(burger).toHaveClass("active");
    });
  });
  describe("Wher it's rendered and cilcked two times on the burger menu", () => {
    test("Then the burger menu should not have class active", () => {
      render(
        <BrowserRouter>
          <Header />
        </BrowserRouter>
      );

      const burger = screen.queryByTitle("Navigation Menu");
      userEvent.dblClick(burger);

      expect(burger).not.toHaveClass("active");
    });
  });
});
