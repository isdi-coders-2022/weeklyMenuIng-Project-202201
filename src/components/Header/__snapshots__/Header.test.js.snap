// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Header Component When it's rendered Then it should match the snapshot 1`] = `
Array [
  <header
    className="sc-hKwDye eIVOiK"
  >
    <img
      alt="My weekly menu logo"
      className="sc-eCImPb czJMsU"
      src="logo.svg"
    />
    <ul>
      <li>
        <a
          href="/home"
          onClick={[Function]}
        >
          My Weekly Menu
        </a>
      </li>
      <li>
        <a
          href="/search"
          onClick={[Function]}
        >
          Search Recipes
        </a>
      </li>
      <li>
        <a
          href="/home/new"
          onClick={[Function]}
        >
          Create New Recipe
        </a>
      </li>
      <li>
        <a
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </li>
    </ul>
    <div
      className="sc-bdvvtL hJvUau burger-container"
      onClick={[Function]}
      title="Navigation Menu"
    >
      <span
        className="sc-gsDKAQ bFRFTX burger-bar"
      />
      <span
        className="sc-gsDKAQ bFRFTX burger-bar"
      />
      <span
        className="sc-gsDKAQ bFRFTX burger-bar"
      />
    </div>
  </header>,
  "",
]
`;
